src/doc/trpl/error-handling.md

```diff
@@ -1,308 +1,2147 @@
-% Error Handling
+# Error Handling
+Like most programming languages, Rust encourages the programmer to handle
+errors in a particular way. Generally speaking, error handling is divided into
+two broad categories: exceptions and return values. Rust opts for return
+values.
+
+In this chapter, we intend to provide a comprehensive treatment of how to deal
+with errors in Rust. More than that, we will attempt to introduce error handling
+one piece at a time so that you'll come away with a solid working knowledge of
+how everything fits together.
+
+When done naively, error handling in Rust can be verbose and annoying. This
+chapter will explore those stumbling blocks and demonstrate how to use the
+standard library to make error handling concise and ergonomic.
+
+## Table of Contents
```

Comment:
I'm more thinking that this chapter having its own TOC is unusual; it is helpful here, but also, more sub-headings in rustbook might be a good thing as well. I dunno :/